src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -82,4 +92,46 @@ public void hideDetails(Submission submission, User user) {
             }
         }
     }
+
+    /**
+     * Creates a new Result object, assigns it to the given submission and stores the changes to the database. Note, that this method is also called for example submissions which
+     * do not have a participation. Therefore, we check if the given submission has a participation and only then update the participation with the new result.
+     *
+     * @param submission the submission for which a new result should be created
+     * @return the newly created result
+     */
+    public Result setNewResult(Submission submission) {
+        Result result = new Result();
+        result.setSubmission(submission);
+        submission.setResult(result);
+        if (submission.getParticipation() != null) {
+            result.setParticipation(submission.getParticipation());
+        }
```

Comment:
There is a `isExampleSubmission` method in the `Submission` class. Besides, isn't the default value for the participation in the result `null` anyway, so it wouldn't really harm to execute the setter in all cases?

Reply:
You are right we can just call it in all the cases. If it is `null` we just set it to `null`